packages/components/src/navigation/index.js

```diff
@@ -57,13 +60,28 @@ export default function Navigation( {
 		}
 	}, [ activeItem, activeMenu ] );
 
+	useEffect( () => {
+		const items = findNavigationItems( children );
+		const menus = findNavigationMenus( children );
+
+		setNavigationItems( items );
+		setNavigationMenus( menus );
+	}, [] );
```

Comment:
Children might change after mount (e.g. dynamically fetched menu items), so we can't use the empty dependency array here.
But children is not a primitive, so using it as a dependency would run `useEffect` at every render, which might be a performance problem.

I'm not sure how to solve this conundrum, though. 🤔